modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpRequestHandler.java

```diff
@@ -51,9 +54,12 @@ class Netty4HttpRequestHandler extends SimpleChannelInboundHandler<HttpPipelined
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, HttpPipelinedRequest httpRequest) {
         final Netty4HttpChannel channel = ctx.channel().attr(Netty4HttpServerTransport.HTTP_CHANNEL_KEY).get();
+        final RestResponse earlyResponse = ctx.channel().attr(Netty4HttpServerTransport.EARLY_RESPONSE).get();
+        final ThreadContext.StoredContext contextToRestore = ctx.channel().attr(Netty4HttpServerTransport.CONTEXT_TO_RESTORE).get();
```

Comment:
This gives me a 'code smell' avoiding using the thread context would be highly desired.

Reply:
Agreed. With the new `header_verifier` step in the pipeline there may be context to carry over further down the pipeline which is why this is here. Before `dispatchRequest` is called in the [last step of the pipeline](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java#L371-L389) (the actual request handler), there is a call to stash the thread context so that the RestController operates in a fresh context. 

The context from the new `header_verifier` step would need to be carried over and I was able to carry it over through the channel context, but there may be better ways of handling it.

Reply:
With an update to the header verifier to bypass the pipeline I don't think this is needed, removed.